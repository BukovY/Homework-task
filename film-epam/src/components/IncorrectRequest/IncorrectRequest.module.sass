.center
  text-align: center
  span
    color: #e7fdd8
  a
    color: gold
    font-size: 20px
    margin-bottom: 10px
    text-decoration: none
  a:hover
    text-shadow: 0px 0px 10px gold