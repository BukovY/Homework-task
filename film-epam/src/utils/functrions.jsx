export const genresIndexToString = (arr, map) => {
  const genresToRender = [];
  for (let i of map) {
    if (arr.indexOf(i.id) !== -1) {
      genresToRender.push(i.name);
    }
  }
  return genresToRender;
};

export const minToTime = (num) => {
  let min = num % 60;
  let hour = (num - min) / 60;
  return String(hour) + ":" + String(min);
};

export const getPhotoCard = (path) => {
  return path === null || path === undefined
    ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    : `https://image.tmdb.org/t/p/w1280/${path}`;
};

export const getPeopleCard = (img) => {
  return img === null || img === undefined
    ? "https://www.themoviedb.org/assets/2/v4/glyphicons/basic/glyphicons-basic-4-user-grey-d8fe957375e70239d6abdd549fd7568c89281b2179b5f4470e2e12895792dfa5.svg"
    : `https://image.tmdb.org/t/p/w500/${img}`;
};

export const getFilmCover = (img) => {
  return img === null || img === undefined
    ? "https://upload.wikimedia.org/wikipedia/commons/6/65/No-Image-Placeholder.svg"
    : `https://image.tmdb.org/t/p/w500/${img}`;
};

export const getPaginationRange = (maxPage) =>
  Array(maxPage)
    .fill(1)
    .reduce((prev, next, index) => {
      return [...prev, index + 1];
    }, []);

export const getIndexLanguage = (lang) => {
  return lang === "EN" ? 0 : lang === "RU" ? 1 : 2;
};

export const matchOnlyNumber = (str) => /^\d+$/.test(str);
