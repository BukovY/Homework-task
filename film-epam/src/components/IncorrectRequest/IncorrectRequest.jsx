import styles from "./IncorrectRequest.module.sass";
import React from "react";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";
import { getIndexLanguage } from "../../utils/functrions";
import { incorrectRequestTranslation } from "../../static/Translation";

const IncorrectRequest = ({ path }) => {
  const { languageSelected } = useSelector((state) => state.app);
  const indLang = getIndexLanguage(languageSelected);
  return (
    <div className={styles.center}>
      <h1>{incorrectRequestTranslation.title[indLang]}</h1>
      <p>{incorrectRequestTranslation.headline[indLang]}</p>
      <p>
        {incorrectRequestTranslation.spanPrefix[indLang]} <span>{path}</span>{" "}
        {incorrectRequestTranslation.spanPostfix[indLang]}
      </p>
      <p>{incorrectRequestTranslation.navDescriptor[indLang]}</p>
      <Link to={"/"}>
        {incorrectRequestTranslation.homepageButton[indLang]}
      </Link>
    </div>
  );
};

export default IncorrectRequest;
