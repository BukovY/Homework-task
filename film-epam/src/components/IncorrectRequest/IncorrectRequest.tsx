import React, { FC } from "react";
import { useHistory } from "react-router-dom";
import { useSelector } from "react-redux";
import { getIndexLanguage } from "../../utils/functrions";
import { incorrectRequestTranslation } from "../../static/Translation";
import Typography from "@material-ui/core/Typography";
import { Button } from "@material-ui/core";
import { makeStyles } from "@material-ui/core/styles";
import { RootState } from "../../redux/store";
import Box from "@material-ui/core/Box";

const useStyles = makeStyles(() => ({
  root: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
  },
}));

type RequestType = {
  path: string;
};

export const IncorrectRequest: FC<RequestType> = ({ path }) => {
  const { languageSelected } = useSelector((state: RootState) => state.app);
  const indLang = getIndexLanguage(languageSelected);
  const texts = {
    title: incorrectRequestTranslation.title[indLang],
    headline: incorrectRequestTranslation.headline[indLang],
    spanPrefix: incorrectRequestTranslation.spanPrefix[indLang],
    spanPostfix: incorrectRequestTranslation.spanPostfix[indLang],
    navDescriptor: incorrectRequestTranslation.navDescriptor[indLang],
    homepageButton: incorrectRequestTranslation.homepageButton[indLang],
  };
  const history = useHistory();
  const goHomepage = () => {
    history.push("/");
  };

  const classes = useStyles();
  return (
    <Box className={classes.root}>
      <Typography variant="h1">{texts.title}</Typography>
      <Typography variant="body1">{texts.headline}</Typography>
      <Typography variant="body1">
        {texts.spanPrefix}
        <Typography display="inline" color="error">
          {` ${path} `}
        </Typography>
        {texts.spanPostfix}
      </Typography>
      <Typography variant="body1">{texts.navDescriptor}</Typography>
      <Button variant="contained" onClick={goHomepage}>
        {texts.homepageButton}
      </Button>
    </Box>
  );
};
