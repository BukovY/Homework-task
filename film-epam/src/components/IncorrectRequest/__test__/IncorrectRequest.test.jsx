import React from "react";
import rerender from "react-test-renderer";
import { IncorrectRequest } from "../IncorrectRequest";
import { render, screen } from "@testing-library/react";
import { wrapperProvider } from "../../../utils/functrions";

describe("Testing <IncorrectRequest/> Component", () => {
  it("IncorrectRequest view correct", () => {
    const tree = rerender
      .create(wrapperProvider(<IncorrectRequest path={"path"} />))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Renedering of custom path", () => {
    render(wrapperProvider(<IncorrectRequest path={"urlSpan"} />));
    expect(screen.getByText(/urlSpan/i)).toBeInTheDocument();
  });
});
