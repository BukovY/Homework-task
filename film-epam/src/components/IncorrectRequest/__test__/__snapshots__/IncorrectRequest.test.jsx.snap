// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <IncorrectRequest/> Component IncorrectRequest view correct 1`] = `
<div
  className="center"
>
  <h1>
    Invalid request
  </h1>
  <p>
    Please check the correctness of the entered request in the browser line.
  </p>
  <p>
    Way after
     
    <span>
      path
    </span>
     
    should only contain numbers.
  </p>
  <p>
    Or return to the main page.
  </p>
  <a
    href="/"
    onClick={[Function]}
  >
    To main
  </a>
</div>
`;
